src/speculators/models/eagle.py

```diff
@@ -532,7 +529,45 @@ def _create_transformer_layer(self) -> nn.Module:
         return layer
 
     def _layernorm_class(self) -> type[nn.Module]:
-        return LlamaRMSNorm
+        return self._architecture_classes[1]
 
     def _transformer_layer_class(self) -> type[nn.Module]:
-        return LlamaDecoderLayer
+        return self._architecture_classes[0]
+
+    def _import_model_classes(self) -> tuple[type, type]:
+        """
+        Get the decoder layer class and layer normalization class for a given config
+        and decoder layer name. Uses the config qualified name to find the corresponding
+        modeling module to import the decoder layer and normalization classes from.
+        """
+        # e.g. LlamaConfig + LlamaDecoderLayer
+        # -> get qualname: transformers.models.llama.configuration_llama.LlamaConfig
+        # -> expected path: transformers.models.llama.modeling_llama.LlamaDecoderLayer
+        # -> expected path: transformers.models.llama.modeling_llama.LlamaRMSNorm
+
+        module_path = self.config.transformer_layer_config.__class__.__module__
+        parts = module_path.split(".")
+        if module_path.startswith("transformers.models.") and len(parts) >= 3:  # noqa: PLR2004
+            model_name = parts[2]
+        else:
+            raise ValueError(
+                "Could not import model classes. Please ensure the config is a valid "
+                "model specific config. i.e. LlamaConfig"
+            )
+
+        # Import the modeling module
+        modeling_module = importlib.import_module(
+            f"transformers.models.{model_name}.modeling_{model_name}"
+        )
+
+        transformer_arch_name = self.config.transformer_layer_architecture
+        decoder_class = getattr(modeling_module, transformer_arch_name)
+        try:
+            layer_norm_class = getattr(
+                modeling_module,
+                f"{transformer_arch_name.removesuffix('DecoderLayer')}RMSNorm",
```

Comment:
Rather than assuming the "{DecoderLayer}RMSNorm" name pattern that would have to exist and may not transfer across model implementations, a pathway that better ensures generality would be to check a regex or multiple regexes against the class list in the module. Something like the following:

```python
classes = dict(inspect.getmembers(module, inspect.isclass))
for pat in [r".*RMSNorm$", r".*Norm$"]:
    for name, cls in classes.items():
        if re.match(pat, name): return cls
```